src/frequenz/client/dispatch/_client.py

```diff
@@ -37,23 +40,43 @@
 )
 
 # pylint: enable=no-name-in-module
+DEFAULT_DISPATCH_PORT = 50051
 
 
-class Client:
+class Client(BaseApiClient[dispatch_pb2_grpc.MicrogridDispatchServiceStub]):
     """Dispatch API client."""
 
     def __init__(
-        self, *, grpc_channel: grpc.aio.Channel, svc_addr: str, key: str
+        self,
+        *,
+        svc_addr: str,
+        key: str,
+        connect: bool = True,
     ) -> None:
         """Initialize the client.
 
         Args:
-            grpc_channel: gRPC channel to use for communication with the API.
             svc_addr: Address of the service to connect to.
             key: API key to use for authentication.
+            connect: Whether to connect to the service immediately.
         """
+        super().__init__(
+            svc_addr,
+            dispatch_pb2_grpc.MicrogridDispatchServiceStub,
+            connect=connect,
+            channel_defaults=ChannelOptions(
+                port=DEFAULT_DISPATCH_PORT,
+                ssl=SslOptions(
+                    enabled=True,
+                    root_certificates=Path(
+                        str(
+                            files("frequenz.client.dispatch").joinpath("certs/root.crt")
```

Comment:
Doesn't this already return a `Path`?

Reply:
Unfortunately, `joinpath` seems to return a `Traversable`

Reply:
```py
>>> import pathlib
>>> p = pathlib.Path("/tmp").joinpath("lala/papap")
>>> p
PosixPath('/tmp/lala/papap')
>>> isinstance(p, pathlib.Path)
True
```
:thinking:

Reply:
And:
```py
>>> p2 = importlib.resources.files("frequenz.sdk")
>>> p2
PosixPath('/home/luca/devel/sdk/src/frequenz/sdk')
>>> isinstance(p2, pathlib.Path)
True
```

So, a `Path` is probably a `Traversable`, but it is still a `Path` :D

Reply:
Oh, but if it is typed as `Traversable` you would have to cast it I guess. Nevermind, all good, nothing to see here.